File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
answer = 0
found = False

def solution(word):
    global answer, found
    answer = 0
    found = False
    
    alpha = ['A','E','I','O','U']
    
    def dfs(current_word):
        global answer, found
        if len(current_word) > 5:
            return
        if current_word == word:
            found = True
            return
        answer += 1
        for i in alpha:
            if not found:
                dfs(current_word + i)
    
    dfs('')
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    let words = ['A', 'E', 'I', 'O', 'U'];
    let found = false;

    const dfs = (currentWord) => {
        if (currentWord.length > 0) {
            answer++;
        }
        
        if (currentWord === word) {
            found = true;
            return;
        }

        if (currentWord.length >= 5) {
            return;
        }

        for (let i = 0; i < words.length; i++) {
            if (!found) {
                dfs(currentWord + words[i]);
            }
        }
    }

    dfs('');
    return answer;
}

